src/Microsoft.AspNetCore.Mvc.Razor.Host/Internal/ViewComponentTagHelperChunkVisitor.cs

```diff
@@ -0,0 +1,228 @@
+                    $"_viewComponentHelper", "");
+
+                // Add constructor.
+                BuildConstructorString(className);
+
+                // Add attributes.
+                BuildAttributeDeclarations(descriptor);
+
+                // Add process method.
+                BuildProcessMethodString(descriptor);
+            }
+        }
+
+        private void BuildConstructorString(string className)
+        {
+            KeyValuePair<string, string> helperPair = new KeyValuePair<string, string>(
+                _context.IViewComponentHelperType,
+                "viewComponentHelper");
+
+            using (Writer.BuildConstructor(
+                "public",
+                className,
+                new List<KeyValuePair<string, string>>() { helperPair }))
+            {
+                Writer.WriteVariableDeclaration("", "_viewComponentHelper", "viewComponentHelper");
+            }
+        }
+
+        private void BuildAttributeDeclarations(TagHelperDescriptor descriptor)
+        {
+            Writer.Write("[")
+              .WriteMethodInvocation(typeof(HtmlAttributeNotBoundAttribute).FullName, false, new string[0])
+              .WriteParameterSeparator()
+              .Write(_context.ViewContextType)
+              .Write("]")
+              .WriteLine();
+
+            Writer.WriteAutoPropertyDeclaration(
+                "public",
+                _context.ViewContextType,
+                _context.ViewContextType);
+
+            foreach (var attribute in descriptor.Attributes)
+            {
+                Writer.WriteAutoPropertyDeclaration("public", attribute.TypeName, attribute.PropertyName);
+            }
+        }
+
+        private void BuildProcessMethodString(TagHelperDescriptor descriptor)
+        {
+            var contextVariable = "context";
+            var outputVariable = "output";
+
+            using (Writer.BuildMethodDeclaration(
+                    $"public override async",
+                    nameof(System.Threading.Tasks.Task),
+                    "ProcessAsync",
+                    new Dictionary<string, string>()
+                    {
+                        { $"{ _context.TagHelpersNamespace }.{nameof(TagHelperContext)}", contextVariable },
+                        { $"{ _context.TagHelpersNamespace }.{nameof(TagHelperOutput)}", outputVariable }
+                    }))
+            {
+
+                Writer.WriteInstanceMethodInvocation(
+                    $"(({_context.IViewContextAwareType})_viewComponentHelper)",
+                    "Contextualize",
+                    new string[] { _context.ViewContextType });
+
+                var viewComponentName = descriptor.PropertyBag[
+                    ViewComponentTagHelperDescriptorConventions.ViewComponentNameKey];
+                var viewComponentParameters = GetParametersObjectString(descriptor);
+                var methodParameters = new string[] { $"\"{viewComponentName}\"", viewComponentParameters };
```

Comment:
Why not just have the above `GetParametersObjectString` do this? You kinda hacked it the way it is because the array is at most always 2 length with the second parameter consisting of allll of the parameter strings.